mesonbuild/dependencies/misc.py

```diff
@@ -97,7 +97,9 @@ def __init__(self, environment, kwargs):
             for name in header_names:
                 if self.clib_compiler.has_header(name, '', self.env, dependencies=[self], disable_cache=True)[0]:
                     self.is_found = True
-                    self.compile_args = self.link_args = self.clib_compiler.openmp_flags()
+                    self.compile_args = self.clib_compiler.openmp_flags()
+                    if self.clib_compiler.get_id() != 'msvc':
+                        self.link_args = self.clib_compiler.openmp_flags()
```

Comment:
I don't think this is the right place to solve this, I think this should be solved the same way that the Thread dependency does, with two separate methods in the compiler classes an `openmp_flags` and `openmp_link_flags`. Just leave the non MSVC compilers returning the values of `openmp_flags()`, and change MSVC to return `[]`

Reply:
That was my first instinct. Then I saw `if self.clib_compiler.get_id() == 'pgi'` a few lines above in the same function and took the easy way out.

I'm glad to know that my instincts were correct, at least. Force pushed with the requested change.